### ISSUE TITLE
Unexplained cancelled requests

### ISSUE THREAD

**Issue by mmmoli** [nhost/nhost#1244]
Hi folks 👋 

We've started seeing cancelled fetch requests in production.

They only happen for our customers in Thailand. Instance is in europe. 

**Situation**
- Local machine based in the EU
- Works: no requests fail.
- Use VPN to pretend to be in other EU destinations. Always works.
![Screenshot 2022-11-29 at 13 48 54](https://user-images.githubusercontent.com/46807/204546763-cd42f729-ded5-4091-811a-ce3f12172a2c.png)

**Unexpected**
- Visit URL from Thailand
- Use VPN to pretend to be outside of the EU. Always fails.
- requests get cancelled. No Errors
![Screenshot 2022-11-29 at 13 49 53](https://user-images.githubusercontent.com/46807/204547096-064add58-ef53-477b-9e11-72e5634d547a.png)

I've rolled back commits. Doesn't look like it's our application code that changed. So, I'm naturally thinking: nhost infra?

Looks like it could be one of two things. Either:
1. User data is inside the EU instance. So nhost is cancelling and failing silently.
2. The round-trip is taking too long, so the server is giving-up early

Any suggestions?

**Comment by elitan** [comment]
It was canceled at 997ms. Is there a 1-second timeout in your GraphQL client?

**Comment by dbarrosop** [comment]
I think we should test without a VPN as a VPN introduces way too many differences/unknowns, most notably:

1. It duplicates the RTT as your traffic has to go from/to the VPN and then from the VPN server it has to go from/to the EU so you are basically doubling the RTT.
2. A VPN may also introduce MTU issues that can make the traffic drop silently along the path.

The only way of testing reliably would be to test with a machine located there, preferably on your customers premises as this could also be a bad proxy or firewall dropping packets. A network pcap would certainly help troubleshooting this issue.

I'd also be quite surprised if this was an RTT issue as the RTT between Thailand and eu-central-1 should be very similar to between India and eu-central-1 (<200ms) and we have a large customer base in there that has never reported issues connecting to eu-central-1. Also, an RTT of <200ms isn't too bad, not great but it shouldn't lead to connectivity issues.

If you can, would you mind asking the users doing a couple of extra tests?

1. From the terminal run `traceroute $subdomain.auth.$region.nhost.run` (change subdomain and region accordingly)
2. Visit `https://$subdomain.auth.$region.nhost.run/v1/version` on the browser and send a screenshot of developer tools->Network

I'd also be happy to jump into a call and help troubleshoot this if necessary.

**Comment by mmmoli** [comment]
Thanks for the input 🙏 

I can confirm, the timeout does seem to be around the 1s mark every time.
This is strange as our request client has no special configuration for this.

@dbarrosop thank you for the deeper insights. I appreciate that you've told me the same thing twice now (on here, and in nhost discord) 🙈 .

Here's the screenshots from Thailand.

![S__21954695](https://user-images.githubusercontent.com/46807/204838284-a2a8a86f-a901-4ff0-8dae-a15b70496805.jpg)

![S__21962756](https://user-images.githubusercontent.com/46807/204838280-41e30c5f-1806-4fbe-a326-f6b776a2570d.jpg)

**Comment by mmmoli** [comment]
Really appreciate your expertise on this @dbarrosop.

**Scenario** 
Set VPN to Thailand.
nhost instance is in europe-1

**Observation**
This again:
![failed](https://user-images.githubusercontent.com/46807/204547096-064add58-ef53-477b-9e11-72e5634d547a.png)

BUT, here's the output from a http-request tool:
![Screenshot 2022-12-02 at 13 54 36](https://user-images.githubusercontent.com/46807/205308481-3796fa77-a02c-46e5-bef2-28c33e3236d8.png)

All requests work. So I guess something is our app is canceling the request?
If you agree, I'll close this issue and report back the fix/gotcha when we find it 💪

**Comment by dbarrosop** [comment]
Yes, feel free to close and reopen if needed. It would be nice to hear back from you once you figure this out in case it is useful for future reference. I'd also suggest trying to isolate individual components to try to discard things one by one; for instance, trying to run your code locally without the browser, with a different library doing the fetch, etc...

**Comment by mmmoli** [comment]
@dbarrosop great advice. Thank you.

I'll report back any findings.

**Comment by mmmoli** [comment]
Investigating…

In my simplest case, I logged-in then made a request for some data. It failed as above.

I noticed that the Nhost React lib was making 3x requests for the auto-sign-in auth token, in parallel.
It seemed odd so I made an example project:

[Preview](https://vitejsvitelfz4yr-rza4-p1sxzqj2--5173.local-credentialless.webcontainer.io/) and [Repo](https://github.com/mmmoli/htch-thai-bug)

**Instructions**
1. Sign-in
2. Open console to Network tab
3. Refresh the page

**Observation**
I see multiple identical token requests.

**Scenario**
- Local machine based in London
- nhost machine in `ap-southeast-1`
- No VPN

![Screenshot 2022-12-06 at 14 23 31](https://user-images.githubusercontent.com/46807/205937630-a84df8a4-e724-4f7b-9ec8-aae75afea8ac.png)

**Question**
Is this expected behaviour?

**Comment by mmmoli** [comment]
I suspect these issues were due to the urql client we were using.

Switched graphql client and effects have seemed to gone.

**Comment by dbarrosop** [comment]
> I suspect these issues were due to the urql client we were using.

Do you mean the multiple requests, the cancellations or both issues?

**Comment by mmmoli** [comment]
Multiple requests might be a React re-render/hook thing

Urql has the concept of a [dedupExchange](https://formidable.com/open-source/urql/docs/api/core/#dedupexchange) that meant we weren't seeing the multiple requests before.

As for the cancellations. It feels like it's something in Urql. We switched to using [GQty](https://gqty.dev/docs/react/fetching-data) and the Cancelled requests went away.

We never got to the bottom of it. I'm posting here for others to follow along.
